apps/meteor/app/livechat/server/api/v1/room.ts

```diff
@@ -59,14 +59,14 @@ API.v1.addRoute('livechat/room', {
 				return API.v1.success({ room, newRoom: false });
 			}
 
-			let agent;
+			let agent: SelectedAgent | undefined;
 			const agentObj = agentId && (await findAgent(agentId));
 			if (agentObj) {
 				if (isAgentWithInfo(agentObj)) {
 					const { username = undefined } = agentObj;
-					agent = { agentId, username };
+					agent = { agentId, username } as SelectedAgent;
 				} else {
-					agent = { agentId };
+					agent = { agentId } as SelectedAgent;
```

Comment:
this looks wrong as well, since `username` is mandatory on `SelectedAgent`.. maybe `SelectedAgent` needs to be modified to allow both use cases on this file, and then the type cast can be removed.